hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.tsx

```diff
@@ -185,9 +187,18 @@ export class DiskUsage extends React.Component<Record<string, object>, IDUState>
           return (subpath.isKey || subpathName === OTHER_PATH_NAME) ? subpathName : subpathName + '/';
         });
 
-        values = subpaths.map(subpath => {
-          return subpath.size / dataSize;
-        });
+        // To avoid NaN Condition NaN will get divide by Zero to avoid map iterations
+        if (dataSize > 0) {
+          values = subpaths.map(subpath => {
+            return subpath.size / dataSize;
+          });
+        }
+
+        // Adding a MIN_BLOCK_SIZE to each value ensures that even the smallest entities are visible on the pie chart.
+        // Note: The percentage and size string calculations remain unchanged.
+        // Zero Size Entities not adding Minimum Block Size Logic so it will not display inside pi chart.
```

Comment:
Just a nit, can we change the comment for future contributors to understand better regarding this change?
```suggestion
        // Adding a MIN_BLOCK_SIZE to non-zero size entities to ensure that even the smallest entities are visible on the pie chart.
        // Note: The percentage and size string calculations remain unchanged.
```

Reply:
@devabhishekpal 
Done changes as suggested by you.